airflow/utils/file.py

```diff
@@ -377,6 +377,10 @@ def iter_airflow_imports(file_path: str) -> Generator[str, None, None]:
     """Find Airflow modules imported in the given file."""
     try:
         parsed = ast.parse(Path(file_path).read_bytes())
+    except ValueError as e:
+        if "source code string cannot contain null bytes" in str(e):
+            log.warning(f"The file {file_path} contains null bytes and cannot be parsed.")
```

Comment:
Instead of trying to parse the exception, it would be easier to do

```python
try:
    data = Path(file_path).read_bytes()
    if 0 in data:
        log.warning(...)
    parsed = ast.parse(data)
except Exception:
    return
```

Also, [do not use an f-string with logging](https://docs.astral.sh/ruff/rules/logging-f-string/).